src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -279,7 +273,7 @@ public static class DependencyGraphRestoreUtility
             cacheContext.SolutionSpec = solutionDgSpec;
             cacheContext.SolutionSpecHash = newDependencyGraphSpecHash;
 
-            if (oldDependencyGraphSpecHash != newDependencyGraphSpecHash)
+            if (forceRestore || (oldDependencyGraphSpecHash != newDependencyGraphSpecHash))
             {
```

Comment:
wouldn't this increase the first restore time in case of force restore? Did you measure time difference?